### ISSUE TITLE
import: self-import should not check data-remote

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#2917]
```
$ dvc -V
0.75.0

# Self-import works just fine. It even does not use data-remote (I didn't push data).
$ dvc import . sources/file.csv --rev tag3

# But it fails when I checkout a revision with no data-remote specified.
# Data-remote was specified after tag1 revision.
$ dvc import . sources/file.csv --rev tag1
Importing 'sources/file.csv (.)' -> 'file.csv'
ERROR: failed to import 'sources/file.csv' from '.'. - No DVC remote is specified in target repository '.'.
```

**Comment by efiop** [comment]
@dmpetrov Duplicate of https://github.com/iterative/dvc/issues/2862 , we have a WIP patch for it, will be solved soon.

**Comment by efiop** [comment]
For the record: the issue there is that we don't use cache dir by default if no remote is set.

**Comment by efiop** [comment]
Ah

**Comment by efiop** [comment]
@dmpetrov Just to clarify: indeed, we shouldn't try to pull from remote when importing from the same repo that we are in, but that logic will automatically apply after #2599 , as we will simply set cache dir as a default remote if it doesn't exist, so we'll just check our cache against our cache.
